pkgs/by-name/gh/ghostty/package.nix

```diff
@@ -0,0 +1,149 @@
+{
+  lib,
+  stdenv,
+  bzip2,
+  callPackage,
+  expat,
+  fetchFromGitHub,
+  fontconfig,
+  freetype,
+  git,
+  glib,
+  gsettings-desktop-schemas,
+  gtk4,
+  harfbuzz,
+  libadwaita,
+  libGL,
+  libpng,
+  libX11,
+  libXcursor,
+  libXi,
+  libXrandr,
+  ncurses,
+  nix-update-script,
+  nixosTests,
+  oniguruma,
+  pandoc,
+  pkg-config,
+  removeReferencesTo,
+  versionCheckHook,
+  wrapGAppsHook4,
+  zig_0_13,
+  zlib,
+}:
+let
+  # Ghostty needs to be built with --release=fast, --release=debug and
+  # --release=safe enable too many runtime safety checks.
+  zig_hook = zig_0_13.hook.overrideAttrs {
+    zig_default_flags = "-Dcpu=baseline -Doptimize=ReleaseFast --color off";
+  };
+in
+stdenv.mkDerivation (finalAttrs: {
+  pname = "ghostty";
+  version = "1.0.0";
+
+  src = fetchFromGitHub {
+    owner = "ghostty-org";
+    repo = "ghostty";
+    tag = "v${finalAttrs.version}";
+    hash = "sha256-AHI1Z4mfgXkNwQA8xYq4tS0/BARbHL7gQUT41vCxQTM=";
+  };
+
+  strictDeps = true;
+
+  nativeBuildInputs = [
+    glib # Required for `glib-compile-schemas`
+    ncurses
+    pandoc
+    pkg-config
+    removeReferencesTo
+    wrapGAppsHook4
+    zig_hook
+  ];
+
+  buildInputs = [
+    bzip2
+    expat
+    fontconfig
+    freetype
+    harfbuzz
+    libadwaita
+    libGL
+    libpng
+    libX11
+    libXcursor
+    libXi
+    libXrandr
+    oniguruma
+    zlib
+  ];
+
+  dontConfigure = true;
+  # doCheck is set to false because unit tests currently fail inside the Nix sandbox.
+  doCheck = false;
```

Comment:
Do *any* of the tests succeed? If so, could we instead exclude the failing ones?

Reply:
~3 (maybe a few more fail) wip to fix upstream.

Reply:
Zig doesn't really provide a way to exclude tests via the CLI, so we'd rather just fix them upstream at a later release

Reply:
> ~3 (maybe a few more fail) wip to fix upstream.

Would you be OK to link to the upstream work? Future devs will probably unironically love finding a "TODO: flip when [URL] is done" comment.

Reply:
https://github.com/ghostty-org/ghostty/pull/3176 is the link.

Reply:
> Do _any_ of the tests succeed? If so, could we instead exclude the failing ones?

There are 6 tests that fail in the nix sandbox.